utils/fix_xc7_carry.py

```diff
@@ -0,0 +1,595 @@
+
+                                      CLE Row
+
++--------------------------------------------------------------------------+
+|                                                                          |
+|                                                                          |
+|                                               +---+                      |
+|                                               |    +                     |
+|                                               |     +                    |
+|                                     +-------->+ O    +                   |
+|              CO CHAIN               |         |       +                  |
+|                                     |         |       +---------------------> xMUX
+|                 ^                   |   +---->+ CO    +                  |
+|                 |                   |   |     |      +                   |
+|                 |                   |   |     |     +                    |
+|       +---------+----------+        |   |     |    +                     |
+|       |                    |        |   |     +---+                      |
+|       |     CARRY ROW      |        |   |                                |
+|  +--->+ S              O   +--------+   |       xOUTMUX                  |
+|       |                    |        |   |                                |
+|       |                    |        +   |                                |
+|  +--->+ DI             CO  +-------+o+--+                                |
+|       |      CI CHAIN      |        +   |                                |
+|       |                    |        |   |                                |
+|       +---------+----------+        |   |       xFFMUX                   |
+|                 ^                   |   |                                |
+|                 |                   |   |     +---+                      |
+|                 +                   |   |     |    +                     |
+|                                     |   +     |     +    +-----------+   |
+|                                     +--+o+--->+ O    +   |           |   |
+|                                         +     |       +  |    xFF    |   |
+|                                         |     |       +->--D----   Q +------> xQ
+|                                         |     |       +  |           |   |
+|                                         +---->+ CO   +   |           |   |
+|                                               |     +    +-----------+   |
+|                                               |    +                     |
+|                                               +---+                      |
+|                                                                          |
+|                                                                          |
++--------------------------------------------------------------------------+
+
+
```

Comment:
This diagram shows that if both the `O` and `CO` outputs are needed, once must pass through the flip flop.

Reply:
Fixed